baseplate/experiments/providers/multi_variant.py

```diff
@@ -0,0 +1,70 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+from .simple_experiment import SimpleExperiment
+
+
+class MultiVariantExperiment(SimpleExperiment):
+    """Basic experiment, handling more than two variants (typically two
+    controls, and multiple treatments).
+    This type of experiment does not guarantee that changing the variant size
+    will not rebucketing existing users.
+
+    Does log bucketing events.
+
+    The config dict is expected to have the following values:
+
+        * **variants**: array of dicts, each containing the keys 'name'
+          and 'size'. Name is the variant name, and size is the fraction of
+          users to bucket into the corresponding variant. Sizes are expressed
+          as a floating point value between 0 and 1.
+        * **bucket_val**: (Optional) Name of the parameter you want to use for
+          bucketing.  This value must be passed to the call to
+          experiment.variant as a keyword argument.  Defaults to "user_id".
+        * **seed**: (Optional) Overrides the seed for this experiment.  If this
+          is not set, a combination of the experiment name, id, and shuffle
+          version is used.
+    """
+
+    @classmethod
+    def from_dict(cls, id, name, owner, start_ts, stop_ts, config,
+                  enabled=True, **kwargs):
+
+        return cls(
+            id=id,
+            name=name,
+            owner=owner,
+            start_ts=start_ts,
+            stop_ts=stop_ts,
+            enabled=enabled,
+            config=config,
+        )
+
+    def _validate_variants(self, variants):
+
+        super(MultiVariantExperiment, self)._validate_variants(variants)
+
+        if len(variants) < 3:
+            raise ValueError("MultiVariant experiments expect two controls "
+                "and at least one variant.")
+
+    def _choose_variant(self, bucket):
+        """Deterministically choose a percentage-based variant. Every call
+        with the same bucket and varaints will result in the same answer
+
+        :param bucket -- an integer bucket representation
+        :param variants -- a dictionary of
+                           <string:variant name>:<float:percentage> pairs.
+        :return string -- the variant name, or None if bucket doesn't fall into
+                          any of the variants
+        """
+
+        current_offset = 0.0
+
+        for variant in self.variants:
+            current_offset += variant['size'] * self.num_buckets
```

Comment:
I'm pretty sure this multiplication is a bug.  If not, it needs a comment.

Reply:
not a bug. Can add documentation. Have to multiply the size of the variant (ex: `0.5` for 50%) by the number of potential buckets to determine whether the assigned bucket falls in the variant's bucket range.